### DISCUSSION TITLE
Block Sync at 0%

### DISCUSSION THREAD

**Discussion by Pointed45** [nervosnetwork/neuron#2470] [Q&A]
**### For several days or months now, I've been trying to synchronize my wallet ... I've tried everything**
https://drive.google.com/file/d/1FhnovwZSLlTTfW5fKDJ17m8DGC7EuXow/view?usp=sharing

________________
You use Neuron, right? What version of Neuron do you have? You need v103.0 or higher (Help - Check for updates in Neuron)

If you already have that version compatible with the hardfork, please check this:

https://github.com/nervosnetwork/neuron/issues/2388

Here is a list of tips as well:

-First start by simply closing Neuron down, reboot PC and reopen Neuron. A lot of times, it can be that simple, specially using Windows.

-If you use MacOS check that the version of your OS is equal to or greater than 10.15.

-If that does not work, please double check that you have between 25G of free space on your PC.

-Then, make sure you're on the newest version, link ⬇️

https://github.com/nervosnetwork/neuron/releases

- If you are using Windows, install vc-dist all in one package (use install_all.bat). https://www.techpowerup.com/download/visual-c-redistributable-runtime-package-all-in-one/

- Check your Windows Firewall settings to make sure ckb.exe is not blocked. Turn off VPN if you use it.

-Try clearing your cache (full rebuild) from the options. (This forces a rebuild but does not require you to download the blockchain data again.) Click on "Setting -> Data".

- Finally try deleting your blockchain data and syncing from the beginning again.
    - Shutdown Neuron and optionally restart your computer.
    - Delete the following folders:
        - C:\Users<username>\AppData\Roaming\Neuron\chains
        - C:\Users<username>\AppData\Roaming\Neuron\ckb_indexer_data
    - Start Neuron.
    _______________________________________

**Comment by yanguoyu** [comment]
Hi @Pointed45, After I view the logs, I found the CKB node was closed after starting it every time. Can you try to run `CKB node` outside `neuron`?

Steps to run a `ckb node` outside `neuron` was elaborated in https://github.com/nervosnetwork/neuron/issues/2458#issuecomment-1200595791.

1. If the `ckb node` outside `neuron` works properly, `neuron` should sync and you can get your fund out

2. If the `ckb node` outside `neuron` doesn't work properly, the full log(without unrelated info) can be used to locate the error.

**Comment by S0rci3r** [comment]
Hello, I have the same problem, more than a week to synchronize !!!! Before this version, everything worked perfectly well. I'm stuck, because I can't move my CKB until it's synchronized... 26% after 3 days... 
![Capture d’écran_2023-01-14_10-38-26](https://user-images.githubusercontent.com/46403833/212465755-a4998aab-17a0-4a0d-8103-15dad1077c13.png)

**Reply by yanguoyu** [reply]
I'm so sorry to hear that, can you create an issue and import debug information for us?

**Reply by S0rci3r** [reply]
impossible, it always tells me "comments too long".
Yet I explain in 3 sentences...

**Reply by Keith-CY** [reply]
> impossible, it always tells me "comments too long". Yet I explain in 3 sentences...

Is `comments too long` an alert message on GitHub? Pasting `logs` in the issue is hard to read so it's prevented. Please attach the compressed log(zip file) in the issue as https://github.com/nervosnetwork/neuron/issues/2570#issue-1531663526 and we can download it and check the content locally.

**Reply by S0rci3r** [reply]
Ok it's done thanks

**Comment by joshyates1980** [comment]
Since July 2021, my approach was (documented here: [https://docs.nervos.org/docs/basics/tools/#neuron-wallet](https://docs.nervos.org/docs/basics/tools/#neuron-wallet))
"run a CKB node yourself and launch Neuron wallet, then Neuron will NOT start the bundled node, but connects to your node instead. "

My main node migration to version 0.106 was successful, but then eventually failed after approximately a month later (documented here: [https://github.com/nervosnetwork/ckb/discussions/3771](https://github.com/nervosnetwork/ckb/discussions/3771))

Now my Neuron wallet will not sync with CKB node. I upgraded Neuron wallet to latest version, which didn't fix my issue.

I troubleshooted on a different computer by importing my wallet seed, run CKB node, and I still cannot get my Neuron wallet to display my balance.  Sync will not work with CKB node.

If I run Neuron by itself, it will attempt to sync the CKB node through Neuron only.
But, I have not been able to see my wallet balance since December.
The latest CKB node broke on one of my computers, but is currently running on my other computer.

Should I only let Neuron work by itself with CKB bundled node?
Should I stop executed CKB Node through the command prompt?

I would like to have access to my CKB balance again with Neuron.

**Reply by Keith-CY** [reply]
Hi @joshyates1980 

A feature `integrated ckb indexer` was introduced in CKB@v0.105.0, and Neuron@v0.106.0 had a change for it.

In the early versions, `ckb-indexer`(https://github.com/nervosnetwork/ckb-indexer) was started by Neuron as an independent service along with ckb-node, then data flowed as `ckb-node => ckb-indexer => neuron`

Since v0.106.0, Neuron adopts the `ckb-indexer` inside `ckb-node`, so `ckb-indexer` is started by `ckb-node` instead of Neuron. The data flow is as before.

But `ckb-indexer` default to be `off`. If `ckb-node` is still launched by `ckb run`, the workflow breaks.

`ckb-indexer` could be enabled by `ckb run --indexer`, or adding `Indexer` in the `module` field of `ckb.toml`

Ref:
merge indexer into ckb node: https://github.com/nervosnetwork/ckb/pull/3609
use integrated ckb-indexer in neuron: https://github.com/Magickbase/neuron-public-issues/issues/71

**Reply by joshyates1980** [reply]
@Keith-CY Thank you for `ckb run --indexer`, for this fixed my issue. Thank you for explanation. I was able to access my balance. But, my DAO deposit was incorrect. There were only 3 deposits listed. I have around 10 total separate deposits. I noticed these 3 deposits have not been withdrawn since the initial deposit and wallet creation from July 2021. My other deposits which came at later dates are not listed. Any ideas?

**Reply by yanguoyu** [reply]
Have the left 7 deposits been withdrawn? If yes, it will show on the `Completed` tab page.
Have your `Neuron` sync to 100%, If not, maybe you should wait for it.
Then if it syncs to 100%, some DAO is also disappeared. You can try to `clean cache` by clicking preferences -> Data -> Clear Cache.

<img width="613" alt="image" src="https://user-images.githubusercontent.com/12881040/216865183-60aaf5a9-9479-46ce-a8f9-0634dac4a9fe.png">

**Reply by joshyates1980** [reply]
@yanguoyu Thank you, the Clear Cache fixed my DAO issue.  All balances are now displayed.
